### PR TITLE
Added a new Plasma Element, refactored some code

### PR DESCRIPTION
**Pull request by KasparasGud**

### CODE REVIEW

File: software/test_project/src/custom_objects/Avatar.java [hunk 1]
```diff
@@ -1,7 +1,6 @@
 package custom_objects;
 
-import java.awt.Color;
-import java.awt.Graphics;
+import java.awt.*;
```
Comment (Lugges991) [inline software/test_project/src/custom_objects/Avatar.java:3]: good, no static imports.

File: software/test_project/src/custom_objects/Avatar.java [hunk 2]
```diff
@@ -77,11 +76,6 @@ private boolean collidingWithMapElement(){
 				&& (toMoveY+getHeight() >= element.getY() 
 				&& toMoveY <= (element.getY()+element.getHeight()))
 				)
```
Comment (Lugges991) [inline software/test_project/src/custom_objects/Avatar.java:78]: commented out code is ok in temporal commits or when it is used for explaining but should not be pushed as final version.

**Review by Lugges991 — APPROVED** [review]
Refactoring is nice and was required, plasma is also cool.
Good job!

File: software/test_project/src/map_builder/ElementWall.java [hunk 3]
```diff
@@ -21,33 +14,23 @@ public ElementWall(int gridX, int gridY, Color elementColor){
...
+		int y = (getGridY()*Constants.MAP_ELEMENT_SIZE) + Constants.WINDOW_MAP_MARGIN+Constants.WINDOW_HEADER_HEIGHT;
+
+		Graphics2D g2d = (Graphics2D)graphics;
+
+		g2d.setColor(getColor());
+		g2d.fillRect(x, y, getWidth(), getHeight());
+
+
+		//Border color is just the main color but with an alpha value
+		Color color = new Color(getColor().getRed(),getColor().getGreen(),getColor().getBlue(),85);
```
Comment (ElectrifyPowr) [inline software/test_project/src/map_builder/ElementWall.java:27]: Maybe consider declaring a variable for the alpha value either on local class level or on global level if this alpha value should also be used for other elements...
```suggestion
                int alpha = 85;
                //...
		Color color = new Color(getColor().getRed(),getColor().getGreen(),getColor().getBlue(),alpha);
```

**Review by ElectrifyPowr — APPROVED** [review]

**Pull request merged**